app/api/resources/user.py

```diff
@@ -508,3 +508,99 @@ def get(cls):
             return messages.USER_NOT_FOUND, HTTPStatus.NOT_FOUND
 
         return dashboard, HTTPStatus.OK
+
+@users_ns.route("user/forgot_password")
+class ForgotPassword(Resource):
+    @classmethod
+    @users_ns.doc(f"forgot password")
+    @users_ns.response(HTTPStatus.OK, f"{messages.PASSWORD_RESET_MAIL_MESSAGE}")
+    @users_ns.response(HTTPStatus.BAD_REQUEST, f"{messages.EMAIL_FIELD_IS_MISSING}"'\n'f"{messages.EMAIL_INPUT_BY_USER_IS_INVALID}")
+    @users_ns.response(HTTPStatus.FORBIDDEN, f"{messages.USER_HAS_NOT_VERIFIED_EMAIL_BEFORE_LOGIN}")
+    @users_ns.response(HTTPStatus.NOT_FOUND, f"{messages.USER_IS_NOT_REGISTERED_IN_THE_SYSTEM}")
+    @users_ns.expect(forgot_password_change_request_data_model, validate=True)
+ 
+    def post(self):
+        """ Updates User's password by sending reset link to email
+ 
+        Args: 
+          email : Verified email of the user
+ 
+        Return:
+          token: a token link/reset link will be shared on verified mail of user
+        """
+ 
+        data = request.json
+ 
+        if not data["email"]:
+            return messages.EMAIL_FIELD_IS_MISSING, HTTPStatus.BAD_REQUEST
+ 
+        is_valid = validate_forgot_password_email(data)
```

Comment:
@isabelcosta  that method is just trying to find out whether email is a valid one or not whereas the lines 534 to 535 is just trying to ensure that the email field is not missing which is i think a required feature under post to ensure its not missing , what do you think ? please suggest what to do then , should i add these 534 to 535 lines under this validating function [since as its name suggest it will just validat the email provided & will not check the missing condition ]